packages/go/src/symbols/go.ts

```diff
@@ -0,0 +1,228 @@
+import {
+  Namekey,
+  OutputDeclarationSpace,
+  OutputMemberSpace,
+  OutputSpace,
+  OutputSymbol,
+  OutputSymbolOptions,
+  track,
+  TrackOpTypes,
+  trigger,
+  TriggerOpTypes,
+  watch,
+} from "@alloy-js/core";
+import { GoScope } from "../scopes/go.js";
+import { PackageSymbol } from "./package.js";
+
+export interface GoSymbolOptions extends OutputSymbolOptions {
+  pointer?: boolean;
+  canExport?: boolean;
+  exported?: boolean;
+}
+
+export type GoSymbolKinds =
+  | "symbol"
+  | "named-type"
+  | "function"
+  | "field"
+  | "package";
+
+function ensureNameExport(
```

Comment:
I feel like we talked about this on discord but I'm not the biggest fan of tracking pointerness in the symbol name like this. Another approach might be that the symbol itself can store a flag for if its holding a pointer, and your member expression component equivalent can dereference when needed based on this metadata. I imagine there are cases when member expression will need to do the opposite and & a value-bearing symbol so you might need that machinery anyway?

Reply:
It would also simplify this logic a lot, and decouple name from pointerness, simplifying the below code.

Reply:
the exportness and pointerness are stored separately, although I couldn't see any good ways of automatically changing the displayed name based on the pointerness so I'm using that function to ensure the name follows the exportness and pointerness whenever anything changes. If there's another way to display a separate name from the original one then that would be good too.